src/app/Dashboard/Charts/ChartCard.tsx

```diff
@@ -0,0 +1,360 @@
+    });
+  }, [setKey]);
+
+  React.useEffect(() => {
+    addSubscription(serviceContext.target.target().subscribe((_) => updateKey()));
+  }, [addSubscription, serviceContext, updateKey]);
+
+  React.useEffect(() => {
+    addSubscription(controllerContext.controller.hasActiveRecording().subscribe(setHasRecording));
+  }, [addSubscription, controllerContext, setHasRecording]);
+
+  React.useEffect(() => {
+    addSubscription(serviceContext.api.grafanaDashboardUrl().subscribe(setDashboardUrl));
+  }, [addSubscription, serviceContext, setDashboardUrl]);
+
+  React.useEffect(() => {
+    if (!dashboardUrl) {
+      return;
+    }
+    const u = new URL('/d-solo/main', dashboardUrl);
+    u.searchParams.append('theme', props.theme);
+    u.searchParams.append('panelId', String(kindToId(props.chartKind)));
+    u.searchParams.append('to', 'now');
+    u.searchParams.append('from', `now-${props.duration}s`);
+    u.searchParams.append('refresh', `${props.period}s`);
+    setChartSrc(u.toString());
+  }, [dashboardUrl, props.theme, props.chartKind, props.duration, props.period, setChartSrc]);
+
+  React.useEffect(() => {
+    addSubscription(controllerContext.controller.attach().subscribe(setIdx));
+  }, [addSubscription, controllerContext]);
+
+  const refresh = React.useCallback(() => {
+    controllerContext.controller.requestRefresh();
+  }, [controllerContext]);
+
+  React.useEffect(() => {
+    refresh();
+    addSubscription(interval(props.period * 1000).subscribe(() => refresh()));
+  }, [addSubscription, props.period, refresh]);
+
+  const popout = React.useCallback(() => {
+    if (chartSrc && dashboardUrl) {
+      window.open(chartSrc, '_blank');
+    }
+  }, [chartSrc, dashboardUrl]);
+
+  const cardStyle = React.useMemo(() => {
+    return {
+      height: getHeight(props.chartKind),
+    };
+  }, [props.chartKind]);
+
+  const expandButton = React.useMemo(() => {
+    return (
+      <>
+        <Button
+          key={0}
+          aria-label={`Expand ${props.chartKind} chart window`}
+          onClick={updateKey}
+          variant="plain"
+          icon={<SyncAltIcon />}
+          isDisabled={!chartSrc || !dashboardUrl}
+        />
+      </>
+    );
+  }, [props.chartKind, updateKey, chartSrc, dashboardUrl]);
+
+  const popoutButton = React.useMemo(() => {
+    return (
+      <>
+        <Button
+          key={1}
+          aria-label={`Pop out ${props.chartKind} chart`}
+          onClick={popout}
+          variant="plain"
+          icon={<ExternalLinkAltIcon />}
+          isDisabled={!chartSrc || !dashboardUrl}
+        />
+      </>
+    );
+  }, [props.chartKind, popout, chartSrc, dashboardUrl]);
+
+  const actions = React.useMemo(() => {
+    const a = props.actions || [];
+    return [expandButton, popoutButton, ...a];
+  }, [props.actions, expandButton, popoutButton]);
+
+  const header = React.useMemo(() => {
+    const isWide = props.span > 4;
+    const style = {
+      marginBottom: isWide ? '-2.8em' : '',
```

Comment:
Maybe a smaller number to leave more space for dragging? `2em` seems like a better fit?